articles/active-directory/develop/quickstart-v2-java-webapp.md

```diff
@@ -91,28 +105,45 @@ To run this sample you will need:
  1. If you use an integrated development environment, open the sample in your favorite IDE (optional).
  1. Open the **application.properties** file, which can be found in *src/main/resources/*.
  1. Replace application properties.
-   1. Find `aad.clientId` and update the value of `Enter_the_Application_Id_here` with the **Application (client) ID** value of the application you registered. 
-   1. Find `aad.authority` and update the value of `Enter_the_Tenant_Name_Here` with the **Directory (tenant) ID** value of the application you registered.
-   1. Find `aad.secretKey` and update the value of `Enter_the_Client_Secret_Here` with the **Client Secret** you created in **Certificates & Secrets** for the application you registered.
+   1. Find `aad.clientId` and update the value of `Enter_the_Application_Id_here` with the **Application (client) ID** value
+   of the application you registered. 
+   1. Find `aad.authority` and update the value of `Enter_the_Tenant_Name_Here` with the **Directory (tenant) ID** value of
+   the application you registered.
+   1. Find `aad.secretKey` and update the value of `Enter_the_Client_Secret_Here` with the **Client Secret** you created in
+   **Certificates & Secrets** for the application you registered.
 
 > [!div renderon="docs"]
 > Where:
 >
 > - `Enter_the_Application_Id_here` - is the Application Id for the application you registered.
-> - `Enter_the_Client_Secret_Here` - is the **Client Secret** you created in **Certificates & Secrets** for the application you registered.
+> - `Enter_the_Client_Secret_Here` - is the **Client Secret** you created in **Certificates & Secrets** for the applicatio
+you registered.
 
 #### Step 4: Run the code sample
-1. Run the code sample, and open a browser and navigate to *http://localhost:8080*.
-1. The front page contains a **sign-in** button. Click on the **sign-in** button to redirect to Azure Active Directory. The user will be prompted for their credentials.  
-1. After successfully authenticating on Azure Active Directory, they will be redirected to *http://localhost:8080/msal4jsamples/secure/aad*. They are officially signed in to the application, and the page should show information for the signed in account. It will also contain buttons for: 
+To run the project, you can either:
+
+- Run it directly from your IDE by using the embedded spring boot server
+- or package it to a WAR file using maven and deploy it a J2EE container solution: 
+
+##### Running from IDE
+1.If you running you web application from an IDE, click on run, then navigate to the home page of the project. 
+For this sample, the standard home page URL is http://localhost:8080
+1. The front page contains a **sign-in** button. Click on the **sign-in** button to redirect to Azure Active Directory. The
+user will be prompted for their credentials.  
```

Comment:
There are multiple places in the document where the sentence is randomly moved to the next line.

Reply:
edited.